dpctl/tensor/_permute_dims.py

```diff
@@ -0,0 +1,36 @@
+#                       Data Parallel Control (dpctl)
+#
+#  Copyright 2020-2021 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import dpctl.tensor as dpt
+
+
+def permute_dims(X, axes):
+    if type(X) is not dpt.usm_ndarray:
+        raise TypeError("Array is not of usm_ndarray type")
+    if not isinstance(axes, tuple):
```

Comment:
I think `axes` of type `list` should also be supported.

Check needs to be added that `axes` is a permutation of `range(0, X.ndim)`.